pilot/pkg/security/model/authentication.go

```diff
@@ -102,15 +103,21 @@ func ConstructSdsSecretConfigWithCustomUds(name, sdsUdsPath string) *tls.SdsSecr
 			InitialFetchTimeout: features.InitialFetchTimeout,
 		},
 	}
+
+	if requestedType == v3.ClusterType || requestedType == v3.ListenerType {
+		// For v3 clusters/listeners, send v3 secrets
+		cfg.SdsConfig.ResourceApiVersion = core.ApiVersion_V3
```

Comment:
Nit: This is the bad part, we can not make sure the requestedType is always the above two as we elvove.

Reply:
Is your concern new types? Or new versions (like v4)?

Reply:
both

Reply:
XDS v4 will not be for over a year, at least. I don't expect any new resources that refer to secrets in the mean time. Note that this code is purely temporary for 1.7. In 1.8, we expect all clients to be XDS v3 and can remove the compatibility code here